modules/db/src/main/CollExt.scala

```diff
@@ -2,177 +2,220 @@ package lila.db
 
 import scala.collection.breakOut
 import scala.collection.generic.CanBuildFrom
+
+import scala.concurrent.ExecutionContext
+
 import scala.util.{ Success, Failure }
 
-import reactivemongo.api._
-import reactivemongo.api.collections.bson.BSONBatchCommands._
-import reactivemongo.api.commands.GetLastError
-import reactivemongo.bson._
-import reactivemongo.core.protocol.MongoWireVersion
+import com.github.ghik.silencer.silent
 
-trait CollExt { self: dsl with QueryBuilderExt =>
+import reactivemongo.bson.{
+  BSONDocument,
+  BSONDocumentHandler,
+  BSONDocumentWriter,
+  BSONDocumentReader
+}
 
-  final implicit class ExtendColl(coll: Coll) {
+import reactivemongo.api.{
+  BSONSerializationPack,
+  Cursor,
+  CollectionProducer,
+  DB,
+  FailoverStrategy,
+  ReadConcern,
+  ReadPreference
+}
+import reactivemongo.api.collections.GenericCollection
+import reactivemongo.api.collections.bson.BSONCollection
+
+final class CollectionExt private[db] (
```

Comment:
Use `CollectionProducer` mechanism instead of "pimp-my-type" implicit conversion